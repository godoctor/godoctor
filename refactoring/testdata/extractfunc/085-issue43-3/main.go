package main

import (
	"fmt"
	"strings"
)

func main() {
	inputString := "bob" + ""
	hasSuffix := false
	// <<<<<extract,12,2,14,2,check,pass
	if strings.Contains(inputString, "Suffix") {
		hasSuffix = true
	}

	fmt.Print("string:", inputString)
	if hasSuffix {
		fmt.Println(" has Suffix")
	} else {
		fmt.Println(" is missing Suffix")
	}
}
