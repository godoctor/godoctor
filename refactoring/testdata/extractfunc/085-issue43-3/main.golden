package main

import (
	"fmt"
	"strings"
)

func main() {
	inputString := "bob" + ""
	hasSuffix := false
	// <<<<<extract,12,2,14,2,check,pass
	hasSuffix = check(inputString)

	fmt.Print("string:", inputString)
	if hasSuffix {
		fmt.Println(" has Suffix")
	} else {
		fmt.Println(" is missing Suffix")
	}
}

func check(inputString string) bool {
	hasSuffix := false
	if strings.Contains(inputString, "Suffix") {
		hasSuffix = true
	}
	return hasSuffix
}
